xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-default/src/main/java/org/xwiki/refactoring/job/XClassDeletingListener.java

```diff
@@ -0,0 +1,161 @@
+        if (!question.getImpactedObjects().isEmpty()) {
+
+            try {
+                if (this.documentAccessBridge.isAdvancedUser(job.getRequest().getProperty("user.reference"))) {
+                    // Conservative choice: we let the user enable the pages to delete.
+                    question.unselectAll();
+                    // The user can modify the question so it could disable some EntitySelection.
+                    // We add a timeout because when a refactoring job is running, it prevents others to run.
+                    // 5 minutes is probably enough for the user to decide if the process should go on.
+                    boolean ack = job.getStatus().ask(question, 5, TimeUnit.MINUTES);
+                    if (!ack) {
+                        // Without any confirmation, we must cancel the operation.
+                        String message = "The question has been asked, however no answer has been received.";
+                        this.logger.warn(message);
+                        CancelableEvent cancelableEvent = (CancelableEvent) event;
+                        cancelableEvent.cancel(message);
+                    }
+                } else {
+                    question.setRefactoringForbidden(true);
+                    // we don't want the user to answer the question,
+                    // but we want to display that his action is forbidden.
+                    boolean ack = job.getStatus().ask(question, 1, TimeUnit.MINUTES);
+                    if (!ack) {
+                        String message = "The question has been canceled because this refactoring is forbidden.";
+                        CancelableEvent cancelableEvent = (CancelableEvent) event;
+                        cancelableEvent.cancel(message);
+                    }
+                }
+            } catch (InterruptedException e) {
+                this.logger.warn("Confirm question has been interrupted.");
+                CancelableEvent cancelableEvent = (CancelableEvent) event;
+                cancelableEvent.cancel("Question has been interrupted.");
+            }
+        }
+    }
+
+    private void checkIfDeleteIsAllowed(EntitySelection entitySelection, XClassBreakingQuestion question)
+    {
+        DocumentReference documentReference = (DocumentReference) entitySelection.getEntityReference();
+        String query = "select distinct doc.fullName from XWikiDocument doc, BaseObject obj where "
+            + "doc.fullName=obj.name and obj.className=:className";
+
+        String serializedReference = serializer.serialize(documentReference);
+        String className = serializedReference.substring(serializedReference.indexOf(":") + 1);
+        try {
+            List<String> results = this.queryManager.createQuery(query, Query.HQL)
+                .bindValue("className", className)
+                .setWiki(documentReference.getWikiReference().getName()).setLimit(10)
```

Comment:
I don't understand the limit, you are going to have missing documents, don't you ? Either you put everything in the question or you use limit(1) just to make sure there is something and you let the question UI gather the rest.

Reply:
> it won't retrieve the objects of the other wikis

Not a problem. It's an official limitation of the classes that it can only be used in the same wiki.